app/code/Magento/CatalogRule/Pricing/Price/CatalogRulePrice.php

```diff
@@ -99,9 +99,9 @@ public function getValue()
                         $this->product->getId()
                     );
                 $this->value = $this->value ? floatval($this->value) : false;
-                if ($this->value) {
-                    $this->value = $this->priceCurrency->convertAndRound($this->value);
-                }
+            }
+            if ($this->value) {
```

Comment:
I see... Thanks, that's exactly what I was asking.

Then
> fix looks ok to me

Integration test would be really valuable to not ever break this place again, but it's up to you, of course, whether invest time in it or not.